### ISSUE TITLE
Error in schema generator

### ISSUE THREAD

**Issue by rsichny** [axnsan12/drf-yasg#304]
getting this error after updating to 1.13.0 (everything was fine with 1.12.1):

```
Traceback (most recent call last):
  File "/lib/python3.6/site-packages/django/core/handlers/exception.py", line 35, in inner
    response = get_response(request)
  File "/lib/python3.6/site-packages/django/core/handlers/base.py", line 128, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/lib/python3.6/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/lib/python3.6/site-packages/django/views/generic/base.py", line 69, in view
    return self.dispatch(request, *args, **kwargs)
  File "/lib/python3.6/site-packages/rest_framework/views.py", line 483, in dispatch
    response = self.handle_exception(exc)
  File "/lib/python3.6/site-packages/rest_framework/views.py", line 443, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/lib/python3.6/site-packages/rest_framework/views.py", line 480, in dispatch
    response = handler(request, *args, **kwargs)
  File "/lib/python3.6/site-packages/drf_yasg/views.py", line 95, in get
    schema = generator.get_schema(request, self.public)
  File "/lib/python3.6/site-packages/drf_yasg/generators.py", line 244, in get_schema
    paths, prefix = self.get_paths(endpoints, components, request, public)
  File "/lib/python3.6/site-packages/drf_yasg/generators.py", line 402, in get_paths
    operation = self.get_operation(view, path, prefix, method, components, request)
  File "/lib/python3.6/site-packages/drf_yasg/generators.py", line 444, in get_operation
    operation = view_inspector.get_operation(operation_keys)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/view.py", line 40, in get_operation
    body = self.get_request_body_parameters(consumes)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/view.py", line 93, in get_request_body_parameters
    schema = self.get_request_body_schema(serializer)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/view.py", line 150, in get_request_body_schema
    return self.serializer_to_schema(serializer)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 438, in serializer_to_schema
    self.field_inspectors, 'get_schema', serializer, {'field_inspectors': self.field_inspectors}
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 118, in probe_inspectors
    result = method(obj, **kwargs)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 39, in get_schema
    return self.probe_field_inspectors(serializer, openapi.Schema, self.use_definitions)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 238, in probe_field_inspectors
    swagger_object_type=swagger_object_type, use_references=use_references, **kwargs
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 118, in probe_inspectors
    result = method(obj, **kwargs)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 125, in field_to_swagger_object
    actual_schema = definitions.setdefault(ref_name, make_schema_definition)
  File "/lib/python3.6/site-packages/drf_yasg/openapi.py", line 679, in setdefault
    ret = maker()
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 101, in make_schema_definition
    child, ChildSwaggerType, use_references, **prop_kwargs
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 238, in probe_field_inspectors
    swagger_object_type=swagger_object_type, use_references=use_references, **kwargs
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 118, in probe_inspectors
    result = method(obj, **kwargs)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 79, in field_to_swagger_object
    child_schema = self.probe_field_inspectors(field.child, ChildSwaggerType, use_references)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 238, in probe_field_inspectors
    swagger_object_type=swagger_object_type, use_references=use_references, **kwargs
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 118, in probe_inspectors
    result = method(obj, **kwargs)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 125, in field_to_swagger_object
    actual_schema = definitions.setdefault(ref_name, make_schema_definition)
  File "/lib/python3.6/site-packages/drf_yasg/openapi.py", line 679, in setdefault
    ret = maker()
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 101, in make_schema_definition
    child, ChildSwaggerType, use_references, **prop_kwargs
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 238, in probe_field_inspectors
    swagger_object_type=swagger_object_type, use_references=use_references, **kwargs
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/base.py", line 118, in probe_inspectors
    result = method(obj, **kwargs)
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 628, in field_to_swagger_object
    values_type = get_basic_type_info_from_hint(next(iter(enum_value_types)))
  File "/lib/python3.6/site-packages/drf_yasg/inspectors/field.py", line 502, in get_basic_type_info_from_hint
    if typing and get_origin_type(hint_class) == typing.Union:
  File "/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  File "/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  File "/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  [Previous line repeated 215 more times]
  File "/lib/python3.6/typing.py", line 759, in __eq__
    if not isinstance(other, _Union):
RecursionError: maximum recursion depth exceeded while calling a Python object
```
Had no chance to dig deeper yet, most likely related to https://github.com/axnsan12/drf-yasg/pull/272

**Comment by axnsan12** [comment]
It would really help if you could reduce this to a simple reproduction.

The last part of the traceback seems really strange.

**Comment by rsichny** [comment]
@axnsan12 it's on my list, will do as soon as i'll get to it

**Comment by rsichny** [comment]
so here's a funny part:
```
Python 3.6.0 (default, Mar 16 2017, 13:31:12) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> t = type('1')
>>> import typing
>>> t == typing.Union
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rs/.pyenv/versions/3.6.0/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  File "/home/rs/.pyenv/versions/3.6.0/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  File "/home/rs/.pyenv/versions/3.6.0/lib/python3.6/typing.py", line 760, in __eq__
    return self._subs_tree() == other
  [Previous line repeated 245 more times]
RecursionError: maximum recursion depth exceeded
```

**Comment by rsichny** [comment]
and this doesn't happen in py 3.7:
```
Python 3.6.7 (default, Feb 17 2019, 12:49:24) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import typing
>>> t = type('1')
>>> t == typing.Union
False
```

The solution is quite simple - type equality should anyway be checked using `is` and not using `==`, that will work for older versions too

**Comment by rsichny** [comment]
@axnsan12 can we please get new minor version deployed after the related PR is merged?

**Comment by axnsan12** [comment]
Sure, will do that on Monday, thanks!

**Comment by axnsan12** [comment]
Ok, so looking more closely at this

> Python 3.6.0 (default, Mar 16 2017, 13:31:12)

> Python 3.6.7 (default, Feb 17 2019, 12:49:24)

Seems to me that this addresses a bug that was fixed in a minor version of Python 3.6. As such, this does not look like something we want to be supporting.

**Comment by rsichny** [comment]
@axnsan12 yeah, but the only reason was in `get_origin_type(hint_class) == typing.Union` vs `get_origin_type(hint_class) is typing.Union` (`is` is a correct usage here anyway). The rest of the fix is related to the early implementation of `typing.Union`, if you don't want to support 3.5.0-3.5.2 i can make it just one line change
